mantle/kola/tests/misc/network.go

```diff
@@ -59,21 +60,30 @@ type listener struct {
 func checkListeners(c cluster.TestCluster, expectedListeners []listener) error {
 	m := c.Machines()[0]
 
-	output := c.MustSSH(m, "sudo netstat -plutn")
+	output := c.MustSSH(m, "sudo ss -plutn")
 
 	processes := strings.Split(string(output), "\n")
 	// verify header is as expected
 	if len(processes) < 2 {
-		c.Fatalf("expected at least two lines of nestat output: %q", output)
+		c.Fatalf("expected at least two lines of ss output: %q", output)
 	}
-	if processes[0] != "Active Internet connections (only servers)" {
-		c.Fatalf("netstat output has changed format: %q", output)
-	}
-	if !regexp.MustCompile(`Proto\s+Recv-Q\s+Send-Q\s+Local Address\s+Foreign Address\s+State\s+PID/Program name`).MatchString(processes[1]) {
-		c.Fatalf("netstat output has changed format: %q", output)
+	// ss output's header sometimes does not have whitespace between "Peer Address:Port" and "Process"
+	headerRegex := `Netid\s+State\s+Recv-Q\s+Send-Q\s+Local Address:Port\s+Peer Address:Port\s*Process`
+	if !regexp.MustCompile(headerRegex).MatchString(processes[0]) {
+		c.Fatalf("ss output has changed format: %q", processes[0])
 	}
 	// skip header
-	processes = processes[2:]
+	processes = processes[1:]
+
+	// create expectedListeners map
+	expectedListenersMap := map[listener]bool{}
+	for _, expected := range expectedListeners {
+		// allow expected tcp to match tcp6
+		expected.protocol = strings.TrimRightFunc(expected.protocol, func(r rune) bool {
+			return unicode.IsNumber(r)
+		})
```

Comment:
Can we simplify this a bit? For my understanding I'll ask some questions. Which one of the following is true:

- we want to be able to specify `tcp` or `tcp6` in the test and have it match the output
- we want to be able to just specify `tcp` in the test and have `tcp` or `tcp6` satisfy the test
So we are able to specify in the test that we want a process/port/protocol and we can specify `tcp` or `tcp6` in the test.

Reply:
Did this to make the line to ensure a similar behaviour as `if strings.HasPrefix(proto, expected.protocol) && // allow expected tcp to match tcp6` in the original code when using a map. 
**However**, the latter of your two statements is true. I misunderstood how `strings.HasPrefix()` works so I messed up the logic. `strings.TrimRightFunc()` should be done to `thisListener.protocol`, instead of `expected.protocol`, my bad.

Reply:
Actually, it looks like `ss` doesn't differentiate between e.g. tcp and tcp6 at the Protocol column, so we should be able to safely remove these lines.